staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -1577,3 +1740,65 @@ func Test_shouldSyncNode(t *testing.T) {
 		})
 	}
 }
+
+func TestTriggerNodeSync(t *testing.T) {
+	controller, _, _ := newController()
+
+	tryReadFromChannel(t, controller.nodeSyncCh, false)
+	controller.triggerNodeSync()
+	tryReadFromChannel(t, controller.nodeSyncCh, true)
+	tryReadFromChannel(t, controller.nodeSyncCh, false)
+	tryReadFromChannel(t, controller.nodeSyncCh, false)
+	tryReadFromChannel(t, controller.nodeSyncCh, false)
+	controller.triggerNodeSync()
+	controller.triggerNodeSync()
+	controller.triggerNodeSync()
+	tryReadFromChannel(t, controller.nodeSyncCh, true)
+	tryReadFromChannel(t, controller.nodeSyncCh, false)
+	tryReadFromChannel(t, controller.nodeSyncCh, false)
+	tryReadFromChannel(t, controller.nodeSyncCh, false)
+}
+
+func tryReadFromChannel(t *testing.T, ch chan interface{}, expectValue bool) {
+	select {
+	case _, ok := <-ch:
+		if !ok {
+			t.Errorf("The channel is closed")
+		}
+		if !expectValue {
+			t.Errorf("Does not expect value from the channel")
```

Comment:
add a "but got a value" to make the error clearer?

Reply:
done